extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +348,94 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
+
+    /** @private {function(!Event):void} */
+    this.debouncedBoundOnSession_ = debounce(
+      this.root.ampdoc.win,
+      this.boundOnSession_,
+      TIME_WAIT
+    );
+    browserEventListeners();
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(BrowserEventType).forEach((key) => {
+      root.removeEventListener(
+        BrowserEventType[key],
+        this.debouncedBoundOnSession_
+      );
+    });
```

Comment:
Since we would know the added listener types, let's remove those only.

```suggestion
    Object.keys(this.hasListener_).forEach((eventName) => {
      root.removeEventListener(
        eventName,
        this.debouncedBoundOnSession_
      );
    });
```

Reply:
Sounds good. Made the changes in the latest commit.